### DISCUSSION TITLE
Zustand V4 createStore/useStore performances within context

### DISCUSSION THREAD

**Discussion by gsouf** [pmndrs/zustand#2268] [Q&A] [answered]
Hi,

I checked the example provided in the [change logs for v4](https://github.com/pmndrs/zustand/releases/tag/v4.0.0) about the new api to make it easy to create a context from zustand to provide dependency injection.

I'm wondering about the performance of the [provided codesandbox example](https://codesandbox.io/s/polished-pond-4jn1e2). 

We can't use the usual zustand state function to pick re-render condition (i.e `const foo = useMyStore(state => state.foo)` <= the component re-renders only when foo changes).

From my understanding it seems that given the way this codesandbox example works then all components/hooks using this context-store will re-render anytime any value in the store changes. 

Is that so or did I miss something?

**Comment by dai-shi** [comment, marked as answer]
You can use the selector like this:
```js
const useMyStore = (selector) => {
  const store = useContext(MyContext);
  if (store === null) {
    throw new Error("no provider");
  }
  return useStore(store, selector);
};
```

**Reply by gsouf** [reply]
oh great! If there isn't any documentation for that yet, I'm pretty sure it would be useful to many to have it documented somewhere, because this way of using zustand solves many use cases!

**Reply by dai-shi** [reply]
Yeah, we have docs for `create`, but for `createStore / useStore` they are probably missing.
@dbritto-dev is it correct?
@charkour so, there is room for improvement in docs.

**Reply by dbritto-dev** [reply]
@dai-shi yeah, it's but seams we have an issue with our search engine since doesn't show anything relevant or anything at all when you type `createStore` or `useStore`. btw, here's the doc -> https://docs.pmnd.rs/zustand/guides/initialize-state-with-props

**Reply by dai-shi** [reply]
Hmm, maybe the search index isn't updated. I'm not very sure.

**Reply by dbritto-dev** [reply]
@dai-shi do you know  if we use something internal? if that the case I would like to fix it

**Reply by dai-shi** [reply]
The code is there: https://github.com/pmndrs/website/tree/docs
AFAIR, it uses Next.js ISR and maybe it's a caching issue?
There's a PR open for a feature, but seems delayed: https://github.com/pmndrs/website/pull/270

**Reply by dbritto-dev** [reply]
May the issue is about how is crawling the docs in order to get the data

**Reply by gsouf** [reply]
I'd like to add, in case it makes any sense to you, that for me the title of the page in the documentation doesn't feel like it's what I'm looking for. So if I was browsing the document searching for this feature it wouldn't catch my attention.

Indeed, it's called "Initialize state with props". But what I'm trying to achieve is to have multiple instances of the same store - like with the multiple count buttons in [the provided codesandbox](https://codesandbox.io/s/polished-pond-4jn1e2), that I can reuse within a react context.

The keywords that make sense to me are "context" and "multiple instances"

**Reply by dai-shi** [reply]
https://docs.pmnd.rs/zustand/guides/initialize-state-with-props is a guide for certain usage.
Instead of modifying it, which serves for different purpose, I think it's better to have a new guide for "context" and " multiple instances". Or, maybe API reference rather than a guide.

Related discussion: https://github.com/pmndrs/zustand/issues/206
